fedbiomed/researcher/secagg/_secure_aggregation.py

```diff
@@ -379,8 +669,195 @@ def load_state_breakpoint(
+
+    This class is responsible for;
+
+    - setting up the context for LOM secure aggregation
+    - Applying secure aggregation after receiving encrypted model parameters from nodes
+
+    Attributes:
+        _dh: Diffie Hellman keypairs per node context setup instance.
+    """
+
+    def __init__(
+            self,
+            active: bool = True,
+            clipping_range: Union[None, int] = None,
+    ) -> None:
+        """Class constructor
+
+        Assigns default values for attributes
+
+        Args:
+            active: True if secure aggregation is activated for the experiment
+            clipping_range: Clipping range that will be used for quantization of model
+                parameters on the node side. The default will be
+                [`VEParameters.CLIPPING_RANGE`][fedbiomed.common.constants.VEParameters].
+                The default value will be automatically set on the node side.
+
+        Raises:
+            FedbiomedSecureAggregationError: bad argument type
+        """
+        super().__init__(active, clipping_range)
+
+        self._dh: Optional[SecaggDhContext] = None
+        self._secagg_crypter: SecaggLomCrypter = SecaggLomCrypter()
+        self._scheme = SecureAggregationSchemes.LOM
+
+    @property
+    def dh(self) -> Union[None, SecaggDhContext]:
+        """Gets Diffie Hellman keypairs object
+
+        Returns:
+            DH object, None if DH is not setup
+        """
+        return self._dh
+
+    def train_arguments(self) -> Dict:
+        """Gets train arguments for secagg train request
+
+        Returns:
+            Arguments that are going to be attached to the experiment.
+        """
+        arguments = super().train_arguments()
+        arguments.update({
+            'secagg_dh_id': self._dh.secagg_id if self._dh is not None else None,
+        })
+        return arguments
+
+    def setup(self,
+              parties: List[str],
+              experiment_id: str,
+              force: bool = False) -> bool:
+        """Setup secure aggregation instruments.
+
+        Requires setting `parties` and `experiment_id` if they are not set in previous secagg
+        setups. It is possible to execute without any argument if SecureAggregation
+        has already `parties` and `experiment_id` defined. This feature provides researcher
+        execute `secagg.setup()` if any connection issue
+
+        Args:
+            parties: Parties that participates secure aggregation
+            experiment_id: The id of the experiment
+            force: Forces secagg setup even context is already existing
+
+        Returns:
+            Status of setup
+
+        Raises
+            FedbiomedSecureAggregationError: Invalid argument type
+        """
+
+        parties = list(
+            filter(lambda x: x != environ["ID"], parties)
+        )
+
+        super().setup(parties, experiment_id, force)
+
+        if self._dh is None:
+            raise FedbiomedSecureAggregationError(
```

Comment:
Hi @ybouilla,

Thank you. I also confirmed with @mvesin that it may be copy paste from previous implementation. I am going to push a fix for that, and for biprime and servkey setup function